Mathlib/Data/BitVec/Lemmas.lean

```diff
@@ -197,6 +197,32 @@ lemma toFin_ofNat (n : ℕ) :
 
 end
 
+/-!
+### Distributivity of `Std.BitVec.toNat`
+-/
+
+section
+variable (x y : BitVec w)
+open BitVec (toNat)
+
+@[simp] lemma toNat_and : (x &&& y).toNat = x.toNat &&& y.toNat := rfl
+@[simp] lemma toNat_or  : (x ||| y).toNat = x.toNat ||| y.toNat := rfl
+@[simp] lemma toNat_xor : (x ^^^ y).toNat = x.toNat ^^^ y.toNat := rfl
+
+/- `Std.BitVec.toNat_add` and `Std.BitVec.toNat_zero` already exists in Std -/
+attribute [simp] Std.BitVec.toNat_add
+
+lemma toNat_mul : (x * y).toNat = (x.toNat * y.toNat) % 2 ^ w           := rfl
+lemma toNat_sub : (x - y).toNat = (x.toNat + (2 ^ w - y.toNat)) % 2 ^ w := rfl
+
+lemma toNat_neg : (-x).toNat = (2 ^ w - x.toNat) % 2 ^ w := by
+  simp only [Neg.neg, BitVec.neg, BitVec.sub_eq, toNat_sub, ofNat_eq_ofNat, toNat_zero, zero_add]
+
+lemma toNat_natCast (n : ℕ) : toNat (n : BitVec w) = n % 2 ^ w := by
```

Comment:
Isn't this https://leanprover-community.github.io/mathlib4_docs/Std/Data/BitVec/Lemmas.html#Std.BitVec.toNat_ofNat

Reply:
Not quite, the linked lemmas talks about `BitVec.ofNat` on the lhs, whereas this lemmas talks about `Nat.cast`. These two expressions are def-eq, though, so they are definitely related.

I don't think we immediately need this result for the main contribution of this PR, so I'd be happy to remove it if the phrasing of the lemma is controversial.

Reply:
If they're defeq, I don't see why we should have both - `Nat.cast` should be the only spelling, imo